core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -244,6 +245,28 @@ static boolean hasProperty(Schema schema, String propertyName) {
     return schema.getObjectProp(propertyName) != null;
   }
 
+  public static int fieldId(Schema currentSchema, Schema parentSchema, Supplier<String> fieldNameGetter) {
```

Comment:
I'd like to avoid needing this method because of the strange input types. I think those show that the way we're traversing the schema isn't quite right, which is why this is complicated.